File: solutions/2026-04-08/solution.java
// Validate IP Address
// Difficulty: Medium
// URL: https://leetcode.com/problems/validate-ip-address/

public String validIPAddress(String IP) {
  String[] parts = IP.split(".");
  if (parts.length != 4) return "Invalid";
  for (String part : parts) {
    if (!part.matches("[0-9]+") || (part.length() > 1 && part.startsWith("0"))) return "Invalid";
    int num = Integer.parseInt(part);
    if (num > 255 || num < 0) return "Invalid";
    if (part.length() > 3) return "Invalid";
  }
  return "Valid";
}

File: solutions/2026-04-08/solution.py
# Validate IP Address
# Difficulty: Medium
# URL: https://leetcode.com/problems/validate-ip-address/

def validIPAddress(IP: str) -> str:
  parts = IP.split('.')
  if len(parts) != 4:
    return "Invalid"
  for part in parts:
    if not part.isdigit() or (len(part) > 1 and part.startswith('0')):
      return "Invalid"
    if int(part) > 255 or int(part) < 0:
      return "Invalid"
    if len(part) > 3:
      return "Invalid"
  return "Valid"